src/mscorlib/shared/System/IO/FileStream.cs

```diff
@@ -110,13 +135,24 @@ public FileStream(SafeFileHandle handle, FileAccess access, int bufferSize, bool
             if (handle.IsAsync.HasValue && isAsync != handle.IsAsync.Value)
                 throw new ArgumentException(SR.Arg_HandleNotAsync, nameof(handle));
 
-            _access = access;
-            _useAsyncIO = isAsync;
             _exposedHandle = true;
             _bufferLength = bufferSize;
-            _fileHandle = handle;
 
-            InitFromHandle(handle);
+            InitFromHandle(handle, access, isAsync);
+        }
+
+        public FileStream(SafeFileHandle handle, FileAccess access, int bufferSize, bool isAsync)
+        {
+            ValidateAndInitFromHandle(handle, access, bufferSize, isAsync);
+
+            // Note: Cleaner to set the following fields in ValidateAndInitFromHandle,
```

Comment:
> out parameters would be an option right?

You could, but it would probably require a comment to explain it. You'd still have to set the field to the passed-in value, which would require passing those three arguments twice (one argument for the input and one for the output. I added the comment per @jkotas feedback.